File: dyeti-frontend/src/types/meal.ts
import { Product } from './product';

export type Meal = {
  id: string;
  orderInDay: number;
  products: MealProduct[];
  recipes: Recipe[];
};

export type MealProduct = {
  product: Product;
  quantity: number;
};

export type Recipe = {
  id: string;
  recipeName: string;
  description: string;
  steps: string[];
};

File: dyeti-backend/src/main/java/ki/agh/dyeti/model/Meal.java
package ki.agh.dyeti.model;

import jakarta.persistence.*;
import java.util.ArrayList;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "meal")
@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Meal {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private int orderInDay;

    @ManyToOne
    @JoinColumn(name = "plan_id")
    private Plan plan;

    @OneToMany(mappedBy = "meal", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<MealProduct> products = new ArrayList<>();

    @OneToMany(mappedBy = "meal", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<Recipe> recipes = new ArrayList<>();
}
